File: marketdata-common/src/main/java/com/handson/dto/Trade.java
package com.handson.dto;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public class Trade {

    public final String code;
    public final int quantity;
    public final double nominal;

    public Trade(String code, int quantity, double nominal) {
        this.code = code;
        this.quantity = quantity;
        this.nominal = nominal;
    }

    public static Trade fromJson(String input) {
        return new Gson().fromJson(input, Trade.class);
    }


    public String toJson() {
        return new Gson().toJson(this);
    }

    @Override
    public String toString() {
        return toJson();
    }
}

File: CSharp/NetCore/MarketData/src/MarketDataCommon/Dto/Trade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace MarketDataCommon.Dto
{
    public class Trade
    {
        public string Code { get; set; }
        public int Quantity { get; set; }
        public double Nominal { get; set; }

        public Trade(string code, int quantity, double nominal)
        {
            Code = code;
            Quantity = quantity;
            Nominal = nominal;
        }

        public static Trade FromJson(string input)
        {
            return JsonConvert.DeserializeObject<Trade>(input);
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this);
        }

        public override string ToString() => ToJson();

    }
}
